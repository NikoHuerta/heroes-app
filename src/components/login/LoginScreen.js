import React from 'react'
import { useNavigate } from "react-router-dom";

export const LoginScreen = () => {
    
    let history = useNavigate();

    const handleClick = () => {
        history("/");
    };
    
    
    
    
    return (
        <div className="container mt-5">
            <h1> Login Screen </h1>
            <hr/>

            <button
            className="btn btn-primary"
            onClick={ handleClick }>
                Login
            </button>
        </div>
    )
}
