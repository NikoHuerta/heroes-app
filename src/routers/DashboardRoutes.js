import React from 'react'
import {
    Routes,
    Route
  } from 'react-router-dom';
import { DcScreen } from '../components/dc/DcScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { Navbar } from '../components/ui/Navbar';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />   
            <div className="container mt-2">
                <Routes>
                    <Route exact path="/marvel" element={ <MarvelScreen /> }/>
                    <Route exact path="/hero/:heroeId/*" element={ <HeroScreen /> }/>
                    <Route exact path="/dc" element={ <DcScreen /> }/>

                    <Route path="*" element={ <MarvelScreen /> } />
                </Routes>
            </div>
        </>
    )
}
