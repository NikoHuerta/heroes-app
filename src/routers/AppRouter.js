import React from 'react'
import {
    BrowserRouter as Router,
    Routes,
    Route
  } from 'react-router-dom';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';

// import { Navbar } from '../components/ui/Navbar';



export const AppRouter = () => {
    return (
    <Router>
      <div>
        {/* <Navbar /> */}

        <Routes>
          <Route exact path="/login" element={ <LoginScreen /> }/>
          <Route exact path="/*" element={ <DashboardRoutes /> }/>
        </Routes>
        
      </div>
    </Router>
    );
}
