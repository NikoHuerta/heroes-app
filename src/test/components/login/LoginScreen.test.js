import { AuthContext } from '../../../auth/authContext';
import {
    Routes,
    Route,
    MemoryRouter
  } from 'react-router-dom';

import { mount } from 'enzyme';
import { types } from '../../../types/types';
import { LoginScreen } from '../../../components/login/LoginScreen';

import { AppRouter } from '../../../routers/AppRouter';
import { PublicRoute } from '../../../routers/PublicRoute';
import { PrivateRoute } from '../../../routers/PrivateRoute';
import { DashboardRoutes } from '../../../routers/DashboardRoutes';


const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({ 
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('Pruebas en <LoginScreen />', () => {
    
    const contextValue = {
        user:{
            logged: false,
        },
        dispatch: jest.fn()
    };

    // const wrapper = mount(<AuthContext.Provider value={ contextValue }>
    //                         <MemoryRouter initialEntries={ ['/'] } />
    //                             <BrowserRouter>
    //                             <div>
    //                                 <Routes>
    //                                 <Route path="/login" element={ 
    //                                             <PublicRoute>
    //                                             <LoginScreen />
    //                                             </PublicRoute> 
    //                                     }/>
    //                                 <Route path="/*" element={ 
    //                                             <PrivateRoute>
    //                                             <DashboardRoutes />  
    //                                             </PrivateRoute>
    //                                     }/>
    //                                 </Routes>
    //                             </div>
    //                             </BrowserRouter>
    //                     </AuthContext.Provider>);
    const wrapper = mount(<AuthContext.Provider value={ contextValue }>
        <MemoryRouter initialEntries={['/login']}>
            <Routes>
                <Route path='/login' element={<LoginScreen />} />
            </Routes>
        </MemoryRouter>    
    </AuthContext.Provider>

    );

    test('Debe de mostrarse correctamente', () => {
       expect(wrapper).toMatchSnapshot();
    });
   

    test('Debe de realizar el dispatch y la navegacion', () => {
        wrapper.find('input').simulate('change', {
            target: {
               name: 'nameLogin',
               value: 'testNameGenerated' 
            }
        });
        // wrapper.find('form').prop('onSubmit')({
        //     preventDefault(){}
        // });
        const handleClick = () => {
            wrapper.find('form').prop('onSubmit')({
                preventDefault(){}
            });    
        }
        
        handleClick();

        const action = {
            type: types.login,
            payload: {
                name: 'testNameGenerated',
                email: 'test@mail.testdom'
            }
        }

        expect(contextValue.dispatch).toHaveBeenCalledWith(action);
        expect(mockNavigate).toHaveBeenCalledWith('/', { replace: true });

        localStorage.setItem('lastPath', '/marvel');
        handleClick();
        expect(mockNavigate).toHaveBeenCalledWith('/marvel', { replace: true });

    });
    
});
