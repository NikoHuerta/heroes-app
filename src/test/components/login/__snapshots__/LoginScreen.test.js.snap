// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> Debe de mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
     Login Screen 
  </h1>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="nameLogin"
      onChange={[Function]}
      placeholder="Enter your name"
      type="text"
      value=""
    />
    <br />
    <button
      className="btn btn-primary"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
